pyuvsim/simsetup.py

```diff
@@ -911,55 +991,174 @@ def initialize_uvdata_from_params(obs_params):
+            Defaults to current directory.
+        kwargs : dictionary
+            Any additional valid UVData attribute to assign to object.
+
+    Returns:
+        UVData object with zeroed data_array
+    """
+
+    if path_out is None:
+        path_out = '.'
+    if yaml_filename is None:
+        yaml_filename = 'obsparam.yaml'
+    if antenna_layout_filename is None:
+        antenna_layout_filename = 'antenna_layout.csv'
+
+    yaml_filename = check_file_exists_and_increment(os.path.join(path_out, yaml_filename))
+    antenna_layout_filename = check_file_exists_and_increment(os.path.join(path_out, antenna_layout_filename))
+
+    antenna_numbers = None
+    if isinstance(array_layout, dict):
+        antenna_numbers = np.fromiter(six.viewkeys(array_layout), dtype=int)
+        antpos_enu = array_layout.values()
+        if antenna_names is None:
+            antenna_names = antenna_numbers.astype('str')
+        if write_files:
+            _write_layout_csv(antenna_layout_filename, antpos_enu, antenna_names, antenna_numbers)
+
+    freq_params = {'Nfreqs': Nfreqs, 'start_freq': start_freq, 'bandwidth': bandwidth, 'freq_array': freq_array, 'channel_width': channel_width}
+    time_params = {'Ntimes': Ntimes, 'start_time': start_time, 'integration_time': integration_time, 'time_array': time_array}
+    selection_params = {'bls': bls, 'redundant_threshold': redundant_threshold, 'antenna_nums': antenna_nums, 'polarizations': polarizations, 'no_autos': no_autos}
+    tele_params = {'telescope_location': repr(telescope_location), 'telescope_name': telescope_name}
+    layout_params = {'antenna_names': antenna_names, 'antenna_numbers': antenna_numbers, 'array_layout': array_layout}
+
+    freq_params = {k: v for k, v in six.iteritems(freq_params) if v is not None}
+    time_params = {k: v for k, v in six.iteritems(time_params) if v is not None}
+    selection_params = {k: v for k, v in six.iteritems(selection_params) if v is not None}
+    tele_params = {k: v for k, v in six.iteritems(tele_params) if v is not None}
+    layout_params = {k: v for k, v in six.iteritems(layout_params) if v is not None}
+
+    uv_obj = UVData()
+
+    valid_param_names = [getattr(uv_obj, param).name for param in uv_obj]
+
+    extra_kwds = {k: v for k, v in six.iteritems(kwargs) if k in valid_param_names}
+
+    param_dict = {
+        'time': time_params,
+        'freq': freq_params,
+        'select': selection_params,
+        'telescope': tele_params
+    }
+
+    param_dict['config_path'] = path_out
+    param_dict['param_file'] = os.path.basename(yaml_filename)
+
+    if write_files:
+        tele_params['array_layout'] = antenna_layout_filename
+        with open(yaml_filename, 'w') as yfile:
+            yaml.dump(param_dict, yfile, default_flow_style=False)
+
+    # If layout and telescope config yaml files are not written out, this will be used
+    # to set the antenna names/numbers/positions.
```

Comment:
This docstring confuses me. Won't the following lines always be used, regardless of whether the file is written?

Reply:
Outdated comment.. I'll delete it.